Fix/Не работает параметр rules в команде exec-rules в случае если включено использование локальных сценариев

Kyrales opened the pull request:
По issue: https://github.com/bia-technologies/precommit4onec/issues/45

Исправлено: Не работает параметр rules в команде exec-rules в случае если включено использование локальных сценариев

+ Добавлены модульные тесты

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Принудительное выполнение явно указанных локальных сценариев через параметр -rules с подробным логированием и фильтрацией.
  * Учет каталога репозитория при разрешении и хранении путей исходных файлов.

* **Tools**
  * Утилиты для корректировки XML‑форм и удаления лишних концевых пробелов.

* **Tests**
  * Тесты, проверяющие принудительное выполнение и поведение локальных сценариев.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Kyrales committed:
Исправлено: Не работает параметр rules в команде exec-rules в случае если включено использование локальных сценариев

# Conflicts:
#	src/Классы/КомандаВыполнитьСценарии.os
#	src/Модули/СценарииОбработки.os

Kyrales committed:
Добавление модульных тестов

Kyrales committed:
Исправление ошибки: если не указаны сценарии, то подхватывать все локальные сценарии

Kyrales committed:
fix: незначительный фикс в сообщении

Kyrales commented on src/Классы/КомандаВыполнитьСценарии.os:
```diff
@@ -59,7 +59,15 @@
+		ЭтоПринудительноеВыполнениеУказанныхПравил = Истина; // Выполнение по явному указанию по -rules
+	Иначе
+		Лог.Отладка("КомандаВыполнитьСценарии.ВыполнитьКоманду(): Перед выполнением СценарииОбработки.ПолучитьСценарииСПараметрамиВыполнения(): ИменаЗагружаемыхСценариев = Пустые " + ИменаЗагружаемыхСценариев);
+	КонецЕсли;
```
Исправил, убрал параметр

theshadowco commented:
@Kyrales 
Нужно обновиться для разрешения конфликтов

Kyrales committed:
Merge branch 'develop' into fix/not_correct_rules_in_local

# Conflicts:
#	src/Классы/КомандаВыполнитьСценарии.os
#	src/Модули/СценарииОбработки.os

Kyrales committed:
Дополнение к merge develop

Kyrales committed:
Адаптация для под актуальный develop

Kyrales committed:
Исправление ошибки в функции КаталогиИсходныхФайлов()

Не учитывала, что может быть передан другой базовый каталог репозитария. Особенно актуально при работе тестов.

Kyrales commented:
> @Kyrales Нужно обновиться для разрешения конфликтов

@theshadowco обновил изменения с учетом develop

theshadowco commented on src/Модули/СценарииОбработки.os:
```diff
@@ -27,11 +27,65 @@
 				Лог.Ошибка("Сценарии из репозитория не загружены т.к. отсутствует каталог %1", ЛокальныйКаталог);
 			Иначе
-				ФайлыЛокальныхСценариев = НайтиФайлы(ФайлПутьКЛокальнымСценариям.ПолноеИмя, "*.os");
+				ФайлыЛокальныхСценариевДляОбработки = НайтиФайлы(ФайлПутьКЛокальнымСценариям.ПолноеИмя, "*.os");
```
нужно отрефакторить - вынести содержимое ветки в отдельный метод

Kyrales committed:
fix: Исправлено определение текущего каталога при принудительном вызове правила

theshadowco commented:
@Kyrales 
Нужно отрефакторить метод, вынеси код ветки 
<img width="748" height="463" alt="image" src="https://github.com/user-attachments/assets/5baf40cc-b62f-4c4e-bd5d-0ba43ff20cb8" />

Kyrales committed:
fix: Рефакторинг кода

По замечанию https://github.com/bia-technologies/precommit4onec/pull/55#issuecomment-3550678428

Kyrales commented:
> @Kyrales Нужно отрефакторить метод, вынеси код ветки 

@theshadowco отрефакторил. Вынес блок кода в отдельную функцию. Когнитивную сложность устранил.

Kyrales committed:
fix: Исправлена ошибка после рефакторинга

Kyrales commented on src/Модули/СценарииОбработки.os:
```diff
@@ -362,4 +370,76 @@
+			
+			КонецЕсли;
+
+		КонецЦикла;
```
Исправлена ошибка
